lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -258,32 +257,31 @@ def indent_multiline(string)
     end
 
     def should_cleanup?
-      @session && !@session.closed? && @cleanup_working_dirs
+      @session && @cleanup_working_dirs
     end
 
-    def session
-      @session ||= begin
-                     @logger.debug("opening session to #{@ssh_user}@#{@host}")
-                     Net::SSH.start(@host, @ssh_user, ssh_options)
-                   end
+    def session(command)
+      @session = true
+      command_out, slave = PTY.open
+      command_in, write = IO.pipe
+      command_pid = spawn(({'SSHPASS' => @ssh_password} if @ssh_password), '/usr/bin/sshpass', '-e', '/usr/bin/ssh', @host, *ssh_options, command, :in => command_in, :out => slave)
+      return command_in, command_out, command_pid, slave, write
     end
 
     def ssh_options
-      ssh_options = {}
-      ssh_options[:port] = @ssh_port if @ssh_port
-      ssh_options[:keys] = [@client_private_key_file] if @client_private_key_file
-      ssh_options[:password] = @ssh_password if @ssh_password
-      ssh_options[:passphrase] = @key_passphrase if @key_passphrase
-      ssh_options[:keys_only] = true
-      # if the host public key is contained in the known_hosts_file,
-      # verify it, otherwise, if missing, import it and continue
-      ssh_options[:verify_host_key] = VERIFY_HOST_KEY
-      ssh_options[:auth_methods] = available_authentication_methods
-      ssh_options[:user_known_hosts_file] = prepare_known_hosts if @host_public_key
-      ssh_options[:number_of_password_prompts] = 1
-      ssh_options[:verbose] = settings[:ssh_log_level]
-      ssh_options[:logger] = Proxy::RemoteExecution::Ssh::LogFilter.new(Proxy::Dynflow::Log.instance)
-      return ssh_options
+      ssh_options = []
+      ssh_options << "-o User=#{@ssh_user}"
```

Comment:
It surprises me that this works. I would have expected that you would need to use
```suggestion
      ssh_options += ["-o", "User=#{@ssh_user}"]
```

Reply:
Yeah, ssh seems to be surprisingly tolerant

Reply:
It works, because of splat operator (`*`) that I used in `spawn` function.

Reply:
That's almost surely not it. Even running `ssh '-o User=foo' target` seems to work